code/modules/reagents/reagent_containers/hypospray.dm

```diff
@@ -182,6 +182,83 @@
 	icon_state = "green"
 	filled_reagents = list("anti_toxin" = 5)
 
+/obj/item/reagent_containers/hypospray/autoinjector/ateopine
+	name = "autoinjector (ateopine)"
+	desc = "A sealed autoinjector system designed for resuscitation of patients whose hearts have stopped."
+	icon_state = "purple"
+	flags = null
+	filled_reagents = list("ateopine" = 5)
+
+/obj/item/reagent_containers/hypospray/autoinjector/ateopine/do_injection(mob/living/carbon/human/H, mob/living/user)
+	. = ..()
+	var/error = can_resus(H)
```

Comment:
If you can inject things other than humans then please make sure to explicitly check that what you injected was indeed a human.

```suggestion
    if(!istype(H)
        return
	var/error = can_resus(H)
	
```